whois-api/src/main/java/net/ripe/db/whois/api/httpserver/RemoteAddressCustomizer.java

```diff
@@ -63,20 +64,34 @@ public ConnectionMetaData getConnectionMetaData() {
                 return new ConnectionMetaData.Wrapper(request.getConnectionMetaData()) {
                     @Override
                     public SocketAddress getRemoteSocketAddress() {
-                        String remoteAddress = stripBrackets(getRemoteAddrFromRequest(request));
-                        final String clientIp = getClientIp(request);
-
-                        if (isTrusted(remoteAddress) && StringUtils.isNotEmpty(clientIp)){
-                            remoteAddress = clientIp;
-                        }
-
-                        return InetSocketAddress.createUnresolved(URLDecoder.decode(remoteAddress), Request.getRemotePort(request));
+                        return InetSocketAddress.createUnresolved(URLDecoder.decode(getRequestHost()), Request.getRemotePort(request));
                     }
 
                     @Override
                     public boolean isSecure() {
                         return HttpScheme.HTTPS.name().equalsIgnoreCase(getScheme());
                     }
+
+                    private String getRequestHost(){
+                        final String remoteAddress = stripBrackets(getRemoteAddrFromRequest(request));
+                        if (isURIValidEncoded()){
```

Comment:
I think that instead of allowing the request to hit the endpoint
we should early return a 400, this changes the current behaviour